components/html-block/html-block.js

```diff
@@ -209,35 +214,51 @@ class HtmlBlock extends RtlMixin(LitElement) {
 
 	firstUpdated(changedProperties) {
 		super.firstUpdated(changedProperties);
+		this._updateContextKeys();
+	}
 
-		if (this._renderContainer) return;
+	render() {
+		const renderContainerClasses = {
+			'd2l-html-block-rendered': true,
+			'd2l-html-block-compact': this.compact,
+			'd2l-html-block-inline': this.inline,
+			'd2l-html-block-no-deferred-rendering': this.noDeferredRendering
+		};
 
-		// The d2l-html-block-rendered class is used to apply CSS outside of the html-block component. Do not change lightly.
-		this.shadowRoot.innerHTML += '<div class="d2l-html-block-rendered'
-			+ `${this.compact ? ' d2l-html-block-compact' : ''}`
-			+ '"></div><slot'
-			+ `${!this.noDeferredRendering ? ' style="display: none"' : ''}`
-			+ '></slot>';
+		const slotClasses = {
+			'd2l-html-block-deferred-rendering': !this.noDeferredRendering
+		};
 
-		this.shadowRoot.querySelector('slot').addEventListener('slotchange', async e => await this._render(e.target));
-		this._renderContainer = this.shadowRoot.querySelector('.d2l-html-block-rendered');
-		this._context = this._contextObserverController ? { ...this._contextObserverController.values } : {};
+		return html`
+			<div class="${classMap(renderContainerClasses)}"></div>
+			<slot class="${classMap(slotClasses)}" @slotchange="${this._handleSlotChange}"></slot>
+		`;
 	}
 
-	updated() {
-		super.updated();
-		if (this._contextObserverController && this._contextObjectHasChanged()) this._render();
+	updated(changedProperties) {
+		super.updated(changedProperties);
+		if (this._contextChanged()) {
+			this._render();
```

Comment:
There's some jank here because (again) things get really complicated if we need to dynamically change MathJax's configuration. So this logic works in a general sense, but once the MathJax context object is set to something valid the first time, subsequent changes won't accomplish anything.

This isn't ideal, but it's not a change in behaviour and I don't want to muddy this PR with additional work to support dynamically changing MathJax's configuration.

Reply:
Sounds good.  Out of curiosity, do we envision there ever being a use case for it?

Reply:
I have a hard time thinking of one. The big reason we have to handle this via a controller is because in FRAs in particular, it's possible for the component to load before ifrau syncs down the attributes from the host page. However, because the MathJax renderer doesn't instantiate MathJax at all if there's no context object present, that case is covered for us and MathJax will init properly when it's first populated.

So... The use case I think we actually care about now isn't impacted by this long-term "caching" of sorts and I'm not sure there's a use case for adjusting the config on the fly.

Reply:
I have a hard time thinking of one. The big reason we have to handle this via a controller is because in FRAs in particular, it's possible for the component to load before ifrau syncs down the attributes from the host page. However, because the MathJax renderer doesn't instantiate MathJax at all if there's no context object present, that case is covered for us and MathJax will init properly when it's first populated.

So... The use case I think we actually care about now isn't impacted by this long-term "caching" of sorts and I'm not sure there's a use case for adjusting the config on the fly.